streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -49,6 +49,9 @@
 /**
  * {@code StreamsBuilder} provide the high-level Kafka Streams DSL to specify a Kafka Streams topology.
  *
+ * It is a requirement that the processing logic (Topology) be defined in a deterministic way.
+ * If different instances build different runtime code logic the resulting behavior may be unexpected.
```

Comment:
"different" for sure, but this implies that one might have an operator the other does not. The observed issue is, that even if both contain the same operator, they might be added in different order (and thus be named differently) to the `Topology`, thus we should stretch that order matters.